### PR TITLE
add larger packet spec

### PR DESCRIPTION
**Pull request by ami-GS**

### CODE REVIEW

File: udp-rsc-offload.md [hunk 1]
```diff
@@ -42,7 +42,9 @@ URO can only be attempted on a batch of packets that meet **all** the following
 - The UDP checksums on pre-coalesced packets must be correct. This means checksum offload must be enabled and set the checksum OOB info.
 - TTL, ToS/ECN, Protocol, and DF bit must match on all packets (IPv4).
 - TC/ECN, FlowLabel, and HopLimit must match, and NextHeader must be UDP (IPv6).
-- The total length of the Single Coalesced Unit (SCU) must not exceed IP max length.
+- When coalesced packet length is larger than max IP length
+  - set NB->DataLength to the coalesced size (up to 0xFFFFFFFF)
+  - set IP total length field to 0
```
Comment (anrossi) [inline udp-rsc-offload.md:47]: UDP also has a length field. That should also be set to zero.

File: udp-rsc-offload.md [hunk 2]
```diff
@@ -42,7 +42,9 @@ URO can only be attempted on a batch of packets that meet **all** the following
 - The UDP checksums on pre-coalesced packets must be correct. This means checksum offload must be enabled and set the checksum OOB info.
 - TTL, ToS/ECN, Protocol, and DF bit must match on all packets (IPv4).
 - TC/ECN, FlowLabel, and HopLimit must match, and NextHeader must be UDP (IPv6).
-- The total length of the Single Coalesced Unit (SCU) must not exceed IP max length.
+- When coalesced packet length is larger than max IP length
```
Comment (anrossi) [inline udp-rsc-offload.md:45]:
```suggestion
- The total length of the Single Coalesced Unit (SCU) is allowed to exceed IP max length.
- When the SCU length is larger than the max IP length:
```

File: udp-rsc-offload.md [hunk 3]
```diff
@@ -42,7 +42,11 @@ URO can only be attempted on a batch of packets that meet **all** the following
 - The UDP checksums on pre-coalesced packets must be correct. This means checksum offload must be enabled and set the checksum OOB info.
 - TTL, ToS/ECN, Protocol, and DF bit must match on all packets (IPv4).
 - TC/ECN, FlowLabel, and HopLimit must match, and NextHeader must be UDP (IPv6).
-- The total length of the Single Coalesced Unit (SCU) must not exceed IP max length.
+- The total length of the Single Coalesced Unit (SCU) is allowed to exceed IP max length.
+- When the SCU length is larger than the max IP length:
+  - set NB->DataLength to the coalesced size (up to 0xFFFFFFFF)
+  - set IP total length field to 0
```
Comment (anrossi) [inline udp-rsc-offload.md:48]: Add another bullet point about the UDP length. If it is greater than the max UDP length, it should be set to zero.
And another that the UDP checksum field should be zero.

**Review by anrossi — APPROVED** [review]

**Review by nibanks — APPROVED** [review]

File: udp-rsc-offload.md [hunk 4]
```diff
@@ -42,7 +42,12 @@ URO can only be attempted on a batch of packets that meet **all** the following
 - The UDP checksums on pre-coalesced packets must be correct. This means checksum offload must be enabled and set the checksum OOB info.
 - TTL, ToS/ECN, Protocol, and DF bit must match on all packets (IPv4).
 - TC/ECN, FlowLabel, and HopLimit must match, and NextHeader must be UDP (IPv6).
-- The total length of the Single Coalesced Unit (SCU) must not exceed IP max length.
+- The total length of the Single Coalesced Unit (SCU) is allowed to exceed IP max length.
+- When the SCU length is larger than the max IP length:
+  - Set NB->DataLength to the coalesced size (up to 0xFFFFFFFF).
+  - Set IP total length field to 0.
+  - Set UDP length field to 0 when UDP length is greater than max UDP length.
+  - Set UDP checksum field to 0.
```
Comment (mtfriesen) [inline udp-rsc-offload.md:50]: Why are these conditional? IMO that raises complexity on both NICs and the stack, but offers no particular benefit. If we're going to support zeros in these fields for URO, why not make it the default? My preference is to say: URO indications _should_ set the IP length, UDP length, and UDP checksum fields to zero. Components handling URO indications _must_ ignore the IP length, UDP length, and UDP checksum fields.

Comment (nibanks) [reply udp-rsc-offload.md:50]: Yeah, you make a good point. I'll put your suggestion in a new PR.

**Pull request merged**